Upstream-Status: Backport

From 716c93ecf8edc5dda092d4fa4d878f333a601b51 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Mon, 12 Nov 2012 14:04:46 +0000
Subject: [PATCH 30/31] Move asm-generic ioctl definitions to
 linux/ioctlent.h.in

* linux/ioctlent.h.in: Add asm-generic ioctl entries from all
linux/*/ioctlent.h.in files.
* linux/bfin/ioctlent.h.in: Remove asm-generic ioctl entries.
* linux/i386/ioctlent.h.in: Likewise.
* linux/powerpc/ioctlent.h.in: Likewise.
* linux/s390/ioctlent.h.in: Likewise.
* linux/sparc/ioctlent.h.in: Likewise.
---
 linux/bfin/ioctlent.h.in    | 78 -------------------------------------------
 linux/i386/ioctlent.h.in    | 77 -------------------------------------------
 linux/ioctlent.h.in         | 80 +++++++++++++++++++++++++++++++++++++++++++++
 linux/powerpc/ioctlent.h.in | 30 -----------------
 linux/s390/ioctlent.h.in    | 76 ------------------------------------------
 linux/sparc/ioctlent.h.in   | 53 ------------------------------
 6 files changed, 80 insertions(+), 314 deletions(-)

diff --git a/linux/bfin/ioctlent.h.in b/linux/bfin/ioctlent.h.in
index 3773544..791d8e3 100644
--- a/linux/bfin/ioctlent.h.in
+++ b/linux/bfin/ioctlent.h.in
@@ -1,86 +1,8 @@
 	{"asm/bfin_sport.h",	"SPORT_IOC_CONFIG",	0x5001},
 	{"asm/bfin_sport.h",	"SPORT_IOC_GET_SYSTEMCLOCK",	0x5002},
 	{"asm/bfin_sport.h",	"SPORT_IOC_SET_BAUDRATE",	0x5003},
-	{"asm-generic/ioctls.h",	"TCGETS",	0x5401},
-	{"asm-generic/ioctls.h",	"TCSETS",	0x5402},
-	{"asm-generic/ioctls.h",	"TCSETSW",	0x5403},
-	{"asm-generic/ioctls.h",	"TCSETSF",	0x5404},
-	{"asm-generic/ioctls.h",	"TCGETA",	0x5405},
-	{"asm-generic/ioctls.h",	"TCSETA",	0x5406},
-	{"asm-generic/ioctls.h",	"TCSETAW",	0x5407},
-	{"asm-generic/ioctls.h",	"TCSETAF",	0x5408},
-	{"asm-generic/ioctls.h",	"TCSBRK",	0x5409},
-	{"asm-generic/ioctls.h",	"TCXONC",	0x540a},
-	{"asm-generic/ioctls.h",	"TCFLSH",	0x540b},
-	{"asm-generic/ioctls.h",	"TIOCEXCL",	0x540c},
-	{"asm-generic/ioctls.h",	"TIOCNXCL",	0x540d},
-	{"asm-generic/ioctls.h",	"TIOCSCTTY",	0x540e},
-	{"asm-generic/ioctls.h",	"TIOCGPGRP",	0x540f},
-	{"asm-generic/ioctls.h",	"TIOCSPGRP",	0x5410},
-	{"asm-generic/ioctls.h",	"TIOCOUTQ",	0x5411},
-	{"asm-generic/ioctls.h",	"TIOCSTI",	0x5412},
-	{"asm-generic/ioctls.h",	"TIOCGWINSZ",	0x5413},
-	{"asm-generic/ioctls.h",	"TIOCSWINSZ",	0x5414},
-	{"asm-generic/ioctls.h",	"TIOCMGET",	0x5415},
-	{"asm-generic/ioctls.h",	"TIOCMBIS",	0x5416},
-	{"asm-generic/ioctls.h",	"TIOCMBIC",	0x5417},
-	{"asm-generic/ioctls.h",	"TIOCMSET",	0x5418},
-	{"asm-generic/ioctls.h",	"TIOCGSOFTCAR",	0x5419},
-	{"asm-generic/ioctls.h",	"TIOCSSOFTCAR",	0x541a},
-	{"asm-generic/ioctls.h",	"FIONREAD",	0x541b},
-	{"asm-generic/ioctls.h",	"TIOCLINUX",	0x541c},
-	{"asm-generic/ioctls.h",	"TIOCCONS",	0x541d},
-	{"asm-generic/ioctls.h",	"TIOCGSERIAL",	0x541e},
-	{"asm-generic/ioctls.h",	"TIOCSSERIAL",	0x541f},
-	{"asm-generic/ioctls.h",	"TIOCPKT",	0x5420},
-	{"asm-generic/ioctls.h",	"FIONBIO",	0x5421},
-	{"asm-generic/ioctls.h",	"TIOCNOTTY",	0x5422},
-	{"asm-generic/ioctls.h",	"TIOCSETD",	0x5423},
-	{"asm-generic/ioctls.h",	"TIOCGETD",	0x5424},
-	{"asm-generic/ioctls.h",	"TCSBRKP",	0x5425},
-	{"asm-generic/ioctls.h",	"TIOCTTYGSTRUCT",	0x5426},
-	{"asm-generic/ioctls.h",	"TIOCSBRK",	0x5427},
-	{"asm-generic/ioctls.h",	"TIOCCBRK",	0x5428},
-	{"asm-generic/ioctls.h",	"TIOCGSID",	0x5429},
-	{"asm-generic/ioctls.h",	"TCGETS2",	0x542a},
-	{"asm-generic/ioctls.h",	"TCSETS2",	0x542b},
-	{"asm-generic/ioctls.h",	"TCSETSW2",	0x542c},
-	{"asm-generic/ioctls.h",	"TCSETSF2",	0x542d},
-	{"asm-generic/ioctls.h",	"TIOCGRS485",	0x542e},
-	{"asm-generic/ioctls.h",	"TIOCSRS485",	0x542f},
-	{"asm-generic/ioctls.h",	"TIOCGPTN",	0x5430},
-	{"asm-generic/ioctls.h",	"TIOCSPTLCK",	0x5431},
-	{"asm-generic/ioctls.h",	"TCGETX",	0x5432},
-	{"asm-generic/ioctls.h",	"TCSETX",	0x5433},
-	{"asm-generic/ioctls.h",	"TCSETXF",	0x5434},
-	{"asm-generic/ioctls.h",	"TCSETXW",	0x5435},
-	{"asm-generic/ioctls.h",	"TIOCSIG",	0x5436},
-	{"asm-generic/ioctls.h",	"FIONCLEX",	0x5450},
-	{"asm-generic/ioctls.h",	"FIOCLEX",	0x5451},
-	{"asm-generic/ioctls.h",	"FIOASYNC",	0x5452},
-	{"asm-generic/ioctls.h",	"TIOCSERCONFIG",	0x5453},
-	{"asm-generic/ioctls.h",	"TIOCSERGWILD",	0x5454},
-	{"asm-generic/ioctls.h",	"TIOCSERSWILD",	0x5455},
-	{"asm-generic/ioctls.h",	"TIOCGLCKTRMIOS",	0x5456},
-	{"asm-generic/ioctls.h",	"TIOCSLCKTRMIOS",	0x5457},
-	{"asm-generic/ioctls.h",	"TIOCSERGSTRUCT",	0x5458},
-	{"asm-generic/ioctls.h",	"TIOCSERGETLSR",	0x5459},
-	{"asm-generic/ioctls.h",	"TIOCSERGETMULTI",	0x545a},
-	{"asm-generic/ioctls.h",	"TIOCSERSETMULTI",	0x545b},
-	{"asm-generic/ioctls.h",	"TIOCMIWAIT",	0x545c},
-	{"asm-generic/ioctls.h",	"TIOCGICOUNT",	0x545d},
 	{"asm/ioctls.h",	"FIOQSIZE",	0x545e},
-	{"asm-generic/ioctls.h",	"TIOCGHAYESESP",	0x545e},
-	{"asm-generic/ioctls.h",	"TIOCSHAYESESP",	0x545f},
-	{"asm-generic/ioctls.h",	"FIOQSIZE",	0x5460},
 	{"asm/bfin_simple_timer.h",	"BFIN_SIMPLE_TIMER_SET_PERIOD",	0x7402},
 	{"asm/bfin_simple_timer.h",	"BFIN_SIMPLE_TIMER_START",	0x7406},
 	{"asm/bfin_simple_timer.h",	"BFIN_SIMPLE_TIMER_STOP",	0x7408},
 	{"asm/bfin_simple_timer.h",	"BFIN_SIMPLE_TIMER_READ",	0x740a},
-	{"asm-generic/sockios.h",	"FIOSETOWN",	0x8901},
-	{"asm-generic/sockios.h",	"SIOCSPGRP",	0x8902},
-	{"asm-generic/sockios.h",	"FIOGETOWN",	0x8903},
-	{"asm-generic/sockios.h",	"SIOCGPGRP",	0x8904},
-	{"asm-generic/sockios.h",	"SIOCATMARK",	0x8905},
-	{"asm-generic/sockios.h",	"SIOCGSTAMP",	0x8906},
-	{"asm-generic/sockios.h",	"SIOCGSTAMPNS",	0x8907},
diff --git a/linux/i386/ioctlent.h.in b/linux/i386/ioctlent.h.in
index 9a90ec8..449ad40 100644
--- a/linux/i386/ioctlent.h.in
+++ b/linux/i386/ioctlent.h.in
@@ -11,82 +11,5 @@
 	{"asm/mtrr.h",	"MTRRIOC_DEL_PAGE_ENTRY",	0x4d07},
 	{"asm/mtrr.h",	"MTRRIOC_GET_PAGE_ENTRY",	0x4d08},
 	{"asm/mtrr.h",	"MTRRIOC_KILL_PAGE_ENTRY",	0x4d09},
-	{"asm-generic/ioctls.h",	"TCGETS",	0x5401},
-	{"asm-generic/ioctls.h",	"TCSETS",	0x5402},
-	{"asm-generic/ioctls.h",	"TCSETSW",	0x5403},
-	{"asm-generic/ioctls.h",	"TCSETSF",	0x5404},
-	{"asm-generic/ioctls.h",	"TCGETA",	0x5405},
-	{"asm-generic/ioctls.h",	"TCSETA",	0x5406},
-	{"asm-generic/ioctls.h",	"TCSETAW",	0x5407},
-	{"asm-generic/ioctls.h",	"TCSETAF",	0x5408},
-	{"asm-generic/ioctls.h",	"TCSBRK",	0x5409},
-	{"asm-generic/ioctls.h",	"TCXONC",	0x540a},
-	{"asm-generic/ioctls.h",	"TCFLSH",	0x540b},
-	{"asm-generic/ioctls.h",	"TIOCEXCL",	0x540c},
-	{"asm-generic/ioctls.h",	"TIOCNXCL",	0x540d},
-	{"asm-generic/ioctls.h",	"TIOCSCTTY",	0x540e},
-	{"asm-generic/ioctls.h",	"TIOCGPGRP",	0x540f},
-	{"asm-generic/ioctls.h",	"TIOCSPGRP",	0x5410},
-	{"asm-generic/ioctls.h",	"TIOCOUTQ",	0x5411},
-	{"asm-generic/ioctls.h",	"TIOCSTI",	0x5412},
-	{"asm-generic/ioctls.h",	"TIOCGWINSZ",	0x5413},
-	{"asm-generic/ioctls.h",	"TIOCSWINSZ",	0x5414},
-	{"asm-generic/ioctls.h",	"TIOCMGET",	0x5415},
-	{"asm-generic/ioctls.h",	"TIOCMBIS",	0x5416},
-	{"asm-generic/ioctls.h",	"TIOCMBIC",	0x5417},
-	{"asm-generic/ioctls.h",	"TIOCMSET",	0x5418},
-	{"asm-generic/ioctls.h",	"TIOCGSOFTCAR",	0x5419},
-	{"asm-generic/ioctls.h",	"TIOCSSOFTCAR",	0x541a},
-	{"asm-generic/ioctls.h",	"FIONREAD",	0x541b},
-	{"asm-generic/ioctls.h",	"TIOCLINUX",	0x541c},
-	{"asm-generic/ioctls.h",	"TIOCCONS",	0x541d},
-	{"asm-generic/ioctls.h",	"TIOCGSERIAL",	0x541e},
-	{"asm-generic/ioctls.h",	"TIOCSSERIAL",	0x541f},
-	{"asm-generic/ioctls.h",	"TIOCPKT",	0x5420},
-	{"asm-generic/ioctls.h",	"FIONBIO",	0x5421},
-	{"asm-generic/ioctls.h",	"TIOCNOTTY",	0x5422},
-	{"asm-generic/ioctls.h",	"TIOCSETD",	0x5423},
-	{"asm-generic/ioctls.h",	"TIOCGETD",	0x5424},
-	{"asm-generic/ioctls.h",	"TCSBRKP",	0x5425},
-	{"asm-generic/ioctls.h",	"TIOCSBRK",	0x5427},
-	{"asm-generic/ioctls.h",	"TIOCCBRK",	0x5428},
-	{"asm-generic/ioctls.h",	"TIOCGSID",	0x5429},
-	{"asm-generic/ioctls.h",	"TCGETS2",	0x542a},
-	{"asm-generic/ioctls.h",	"TCSETS2",	0x542b},
-	{"asm-generic/ioctls.h",	"TCSETSW2",	0x542c},
-	{"asm-generic/ioctls.h",	"TCSETSF2",	0x542d},
-	{"asm-generic/ioctls.h",	"TIOCGRS485",	0x542e},
-	{"asm-generic/ioctls.h",	"TIOCSRS485",	0x542f},
-	{"asm-generic/ioctls.h",	"TIOCGPTN",	0x5430},
-	{"asm-generic/ioctls.h",	"TIOCSPTLCK",	0x5431},
-	{"asm-generic/ioctls.h",	"TCGETX",	0x5432},
-	{"asm-generic/ioctls.h",	"TIOCGDEV",	0x5432},
-	{"asm-generic/ioctls.h",	"TCSETX",	0x5433},
-	{"asm-generic/ioctls.h",	"TCSETXF",	0x5434},
-	{"asm-generic/ioctls.h",	"TCSETXW",	0x5435},
-	{"asm-generic/ioctls.h",	"TIOCSIG",	0x5436},
-	{"asm-generic/ioctls.h",	"TIOCVHANGUP",	0x5437},
-	{"asm-generic/ioctls.h",	"FIONCLEX",	0x5450},
-	{"asm-generic/ioctls.h",	"FIOCLEX",	0x5451},
-	{"asm-generic/ioctls.h",	"FIOASYNC",	0x5452},
-	{"asm-generic/ioctls.h",	"TIOCSERCONFIG",	0x5453},
-	{"asm-generic/ioctls.h",	"TIOCSERGWILD",	0x5454},
-	{"asm-generic/ioctls.h",	"TIOCSERSWILD",	0x5455},
-	{"asm-generic/ioctls.h",	"TIOCGLCKTRMIOS",	0x5456},
-	{"asm-generic/ioctls.h",	"TIOCSLCKTRMIOS",	0x5457},
-	{"asm-generic/ioctls.h",	"TIOCSERGSTRUCT",	0x5458},
-	{"asm-generic/ioctls.h",	"TIOCSERGETLSR",	0x5459},
-	{"asm-generic/ioctls.h",	"TIOCSERGETMULTI",	0x545a},
-	{"asm-generic/ioctls.h",	"TIOCSERSETMULTI",	0x545b},
-	{"asm-generic/ioctls.h",	"TIOCMIWAIT",	0x545c},
-	{"asm-generic/ioctls.h",	"TIOCGICOUNT",	0x545d},
-	{"asm-generic/ioctls.h",	"FIOQSIZE",	0x5460},
 	{"asm/msr.h",	"X86_IOC_RDMSR_REGS",	0x63a0},
 	{"asm/msr.h",	"X86_IOC_WRMSR_REGS",	0x63a1},
-	{"asm-generic/sockios.h",	"FIOSETOWN",	0x8901},
-	{"asm-generic/sockios.h",	"SIOCSPGRP",	0x8902},
-	{"asm-generic/sockios.h",	"FIOGETOWN",	0x8903},
-	{"asm-generic/sockios.h",	"SIOCGPGRP",	0x8904},
-	{"asm-generic/sockios.h",	"SIOCATMARK",	0x8905},
-	{"asm-generic/sockios.h",	"SIOCGSTAMP",	0x8906},
-	{"asm-generic/sockios.h",	"SIOCGSTAMPNS",	0x8907},
diff --git a/linux/ioctlent.h.in b/linux/ioctlent.h.in
index 895070c..8843b47 100644
--- a/linux/ioctlent.h.in
+++ b/linux/ioctlent.h.in
@@ -734,21 +734,94 @@
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_PVERSION",	0x5400},
 	{"linux/soundcard.h",	"SNDCTL_TMR_TIMEBASE",	0x5401},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_NEXT_DEVICE",	0x5401},
+	{"asm-generic/ioctls.h",	"TCGETS",	0x5401},
 	{"linux/soundcard.h",	"SNDCTL_TMR_START",	0x5402},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_TREAD",	0x5402},
+	{"asm-generic/ioctls.h",	"TCSETS",	0x5402},
 	{"linux/soundcard.h",	"SNDCTL_TMR_STOP",	0x5403},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_GINFO",	0x5403},
+	{"asm-generic/ioctls.h",	"TCSETSW",	0x5403},
 	{"linux/soundcard.h",	"SNDCTL_TMR_CONTINUE",	0x5404},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_GPARAMS",	0x5404},
+	{"asm-generic/ioctls.h",	"TCSETSF",	0x5404},
 	{"linux/soundcard.h",	"SNDCTL_TMR_TEMPO",	0x5405},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_GSTATUS",	0x5405},
+	{"asm-generic/ioctls.h",	"TCGETA",	0x5405},
 	{"linux/soundcard.h",	"SNDCTL_TMR_SOURCE",	0x5406},
+	{"asm-generic/ioctls.h",	"TCSETA",	0x5406},
 	{"linux/soundcard.h",	"SNDCTL_TMR_METRONOME",	0x5407},
+	{"asm-generic/ioctls.h",	"TCSETAW",	0x5407},
 	{"linux/soundcard.h",	"SNDCTL_TMR_SELECT",	0x5408},
+	{"asm-generic/ioctls.h",	"TCSETAF",	0x5408},
+	{"asm-generic/ioctls.h",	"TCSBRK",	0x5409},
+	{"asm-generic/ioctls.h",	"TCXONC",	0x540a},
+	{"asm-generic/ioctls.h",	"TCFLSH",	0x540b},
+	{"asm-generic/ioctls.h",	"TIOCEXCL",	0x540c},
+	{"asm-generic/ioctls.h",	"TIOCNXCL",	0x540d},
+	{"asm-generic/ioctls.h",	"TIOCSCTTY",	0x540e},
+	{"asm-generic/ioctls.h",	"TIOCGPGRP",	0x540f},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_SELECT",	0x5410},
+	{"asm-generic/ioctls.h",	"TIOCSPGRP",	0x5410},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_INFO",	0x5411},
+	{"asm-generic/ioctls.h",	"TIOCOUTQ",	0x5411},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_PARAMS",	0x5412},
+	{"asm-generic/ioctls.h",	"TIOCSTI",	0x5412},
+	{"asm-generic/ioctls.h",	"TIOCGWINSZ",	0x5413},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_STATUS",	0x5414},
+	{"asm-generic/ioctls.h",	"TIOCSWINSZ",	0x5414},
+	{"asm-generic/ioctls.h",	"TIOCMGET",	0x5415},
+	{"asm-generic/ioctls.h",	"TIOCMBIS",	0x5416},
+	{"asm-generic/ioctls.h",	"TIOCMBIC",	0x5417},
+	{"asm-generic/ioctls.h",	"TIOCMSET",	0x5418},
+	{"asm-generic/ioctls.h",	"TIOCGSOFTCAR",	0x5419},
+	{"asm-generic/ioctls.h",	"TIOCSSOFTCAR",	0x541a},
+	{"asm-generic/ioctls.h",	"FIONREAD",	0x541b},
+	{"asm-generic/ioctls.h",	"TIOCLINUX",	0x541c},
+	{"asm-generic/ioctls.h",	"TIOCCONS",	0x541d},
+	{"asm-generic/ioctls.h",	"TIOCGSERIAL",	0x541e},
+	{"asm-generic/ioctls.h",	"TIOCSSERIAL",	0x541f},
+	{"asm-generic/ioctls.h",	"TIOCPKT",	0x5420},
+	{"asm-generic/ioctls.h",	"FIONBIO",	0x5421},
+	{"asm-generic/ioctls.h",	"TIOCNOTTY",	0x5422},
+	{"asm-generic/ioctls.h",	"TIOCSETD",	0x5423},
+	{"asm-generic/ioctls.h",	"TIOCGETD",	0x5424},
+	{"asm-generic/ioctls.h",	"TCSBRKP",	0x5425},
+	{"asm-generic/ioctls.h",	"TIOCTTYGSTRUCT",	0x5426},
+	{"asm-generic/ioctls.h",	"TIOCSBRK",	0x5427},
+	{"asm-generic/ioctls.h",	"TIOCCBRK",	0x5428},
+	{"asm-generic/ioctls.h",	"TIOCGSID",	0x5429},
+	{"asm-generic/ioctls.h",	"TCGETS2",	0x542a},
+	{"asm-generic/ioctls.h",	"TCSETS2",	0x542b},
+	{"asm-generic/ioctls.h",	"TCSETSW2",	0x542c},
+	{"asm-generic/ioctls.h",	"TCSETSF2",	0x542d},
+	{"asm-generic/ioctls.h",	"TIOCGRS485",	0x542e},
+	{"asm-generic/ioctls.h",	"TIOCSRS485",	0x542f},
+	{"asm-generic/ioctls.h",	"TIOCGPTN",	0x5430},
+	{"asm-generic/ioctls.h",	"TIOCSPTLCK",	0x5431},
+	{"asm-generic/ioctls.h",	"TCGETX",	0x5432},
+	{"asm-generic/ioctls.h",	"TIOCGDEV",	0x5432},
+	{"asm-generic/ioctls.h",	"TCSETX",	0x5433},
+	{"asm-generic/ioctls.h",	"TCSETXF",	0x5434},
+	{"asm-generic/ioctls.h",	"TCSETXW",	0x5435},
+	{"asm-generic/ioctls.h",	"TIOCSIG",	0x5436},
+	{"asm-generic/ioctls.h",	"TIOCVHANGUP",	0x5437},
+	{"asm-generic/ioctls.h",	"FIONCLEX",	0x5450},
+	{"asm-generic/ioctls.h",	"FIOCLEX",	0x5451},
+	{"asm-generic/ioctls.h",	"FIOASYNC",	0x5452},
+	{"asm-generic/ioctls.h",	"TIOCSERCONFIG",	0x5453},
+	{"asm-generic/ioctls.h",	"TIOCSERGWILD",	0x5454},
+	{"asm-generic/ioctls.h",	"TIOCSERSWILD",	0x5455},
+	{"asm-generic/ioctls.h",	"TIOCGLCKTRMIOS",	0x5456},
+	{"asm-generic/ioctls.h",	"TIOCSLCKTRMIOS",	0x5457},
+	{"asm-generic/ioctls.h",	"TIOCSERGSTRUCT",	0x5458},
+	{"asm-generic/ioctls.h",	"TIOCSERGETLSR",	0x5459},
+	{"asm-generic/ioctls.h",	"TIOCSERGETMULTI",	0x545a},
+	{"asm-generic/ioctls.h",	"TIOCSERSETMULTI",	0x545b},
+	{"asm-generic/ioctls.h",	"TIOCMIWAIT",	0x545c},
+	{"asm-generic/ioctls.h",	"TIOCGICOUNT",	0x545d},
+	{"asm-generic/ioctls.h",	"TIOCGHAYESESP",	0x545e},
+	{"asm-generic/ioctls.h",	"TIOCSHAYESESP",	0x545f},
+	{"asm-generic/ioctls.h",	"FIOQSIZE",	0x5460},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_START",	0x54a0},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_STOP",	0x54a1},
 	{"sound/asound.h",	"SNDRV_TIMER_IOCTL_CONTINUE",	0x54a2},
@@ -1606,6 +1679,13 @@
 	{"linux/media.h",	"MEDIA_IOC_ENUM_ENTITIES",	0x7c01},
 	{"linux/media.h",	"MEDIA_IOC_ENUM_LINKS",	0x7c02},
 	{"linux/media.h",	"MEDIA_IOC_SETUP_LINK",	0x7c03},
+	{"asm-generic/sockios.h",	"FIOSETOWN",	0x8901},
+	{"asm-generic/sockios.h",	"SIOCSPGRP",	0x8902},
+	{"asm-generic/sockios.h",	"FIOGETOWN",	0x8903},
+	{"asm-generic/sockios.h",	"SIOCGPGRP",	0x8904},
+	{"asm-generic/sockios.h",	"SIOCATMARK",	0x8905},
+	{"asm-generic/sockios.h",	"SIOCGSTAMP",	0x8906},
+	{"asm-generic/sockios.h",	"SIOCGSTAMPNS",	0x8907},
 	{"linux/sockios.h",	"SIOCADDRT",	0x890b},
 	{"linux/sockios.h",	"SIOCDELRT",	0x890c},
 	{"linux/sockios.h",	"SIOCRTMSG",	0x890d},
diff --git a/linux/powerpc/ioctlent.h.in b/linux/powerpc/ioctlent.h.in
index 0870704..510f5aa 100644
--- a/linux/powerpc/ioctlent.h.in
+++ b/linux/powerpc/ioctlent.h.in
@@ -1,36 +1,18 @@
-	{"asm-generic/ioctls.h",	"TCGETS",	0x5401},
-	{"asm-generic/ioctls.h",	"TCSETS",	0x5402},
-	{"asm-generic/ioctls.h",	"TCSETSW",	0x5403},
-	{"asm-generic/ioctls.h",	"TCSETSF",	0x5404},
-	{"asm-generic/ioctls.h",	"TCGETA",	0x5405},
-	{"asm-generic/ioctls.h",	"TCSETA",	0x5406},
-	{"asm-generic/ioctls.h",	"TCSETAW",	0x5407},
-	{"asm-generic/ioctls.h",	"TCSETAF",	0x5408},
-	{"asm-generic/ioctls.h",	"TCSBRK",	0x5409},
-	{"asm-generic/ioctls.h",	"TCXONC",	0x540a},
-	{"asm-generic/ioctls.h",	"TCFLSH",	0x540b},
 	{"asm/ioctls.h",	"TIOCEXCL",	0x540c},
 	{"asm/ioctls.h",	"TIOCNXCL",	0x540d},
 	{"asm/ioctls.h",	"TIOCSCTTY",	0x540e},
-	{"asm-generic/ioctls.h",	"TIOCGPGRP",	0x540f},
-	{"asm-generic/ioctls.h",	"TIOCSPGRP",	0x5410},
-	{"asm-generic/ioctls.h",	"TIOCOUTQ",	0x5411},
 	{"asm/ioctls.h",	"TIOCSTI",	0x5412},
-	{"asm-generic/ioctls.h",	"TIOCGWINSZ",	0x5413},
-	{"asm-generic/ioctls.h",	"TIOCSWINSZ",	0x5414},
 	{"asm/ioctls.h",	"TIOCMGET",	0x5415},
 	{"asm/ioctls.h",	"TIOCMBIS",	0x5416},
 	{"asm/ioctls.h",	"TIOCMBIC",	0x5417},
 	{"asm/ioctls.h",	"TIOCMSET",	0x5418},
 	{"asm/ioctls.h",	"TIOCGSOFTCAR",	0x5419},
 	{"asm/ioctls.h",	"TIOCSSOFTCAR",	0x541a},
-	{"asm-generic/ioctls.h",	"FIONREAD",	0x541b},
 	{"asm/ioctls.h",	"TIOCLINUX",	0x541c},
 	{"asm/ioctls.h",	"TIOCCONS",	0x541d},
 	{"asm/ioctls.h",	"TIOCGSERIAL",	0x541e},
 	{"asm/ioctls.h",	"TIOCSSERIAL",	0x541f},
 	{"asm/ioctls.h",	"TIOCPKT",	0x5420},
-	{"asm-generic/ioctls.h",	"FIONBIO",	0x5421},
 	{"asm/ioctls.h",	"TIOCNOTTY",	0x5422},
 	{"asm/ioctls.h",	"TIOCSETD",	0x5423},
 	{"asm/ioctls.h",	"TIOCGETD",	0x5424},
@@ -38,22 +20,11 @@
 	{"asm/ioctls.h",	"TIOCSBRK",	0x5427},
 	{"asm/ioctls.h",	"TIOCCBRK",	0x5428},
 	{"asm/ioctls.h",	"TIOCGSID",	0x5429},
-	{"asm-generic/ioctls.h",	"TCGETS2",	0x542a},
-	{"asm-generic/ioctls.h",	"TCSETS2",	0x542b},
-	{"asm-generic/ioctls.h",	"TCSETSW2",	0x542c},
-	{"asm-generic/ioctls.h",	"TCSETSF2",	0x542d},
 	{"asm/ioctls.h",	"TIOCGRS485",	0x542e},
 	{"asm/ioctls.h",	"TIOCSRS485",	0x542f},
 	{"asm/ioctls.h",	"TIOCGPTN",	0x5430},
 	{"asm/ioctls.h",	"TIOCSPTLCK",	0x5431},
-	{"asm-generic/ioctls.h",	"TCGETX",	0x5432},
-	{"asm-generic/ioctls.h",	"TCSETX",	0x5433},
-	{"asm-generic/ioctls.h",	"TCSETXF",	0x5434},
-	{"asm-generic/ioctls.h",	"TCSETXW",	0x5435},
 	{"asm/ioctls.h",	"TIOCSIG",	0x5436},
-	{"asm-generic/ioctls.h",	"FIONCLEX",	0x5450},
-	{"asm-generic/ioctls.h",	"FIOCLEX",	0x5451},
-	{"asm-generic/ioctls.h",	"FIOASYNC",	0x5452},
 	{"asm/ioctls.h",	"TIOCSERCONFIG",	0x5453},
 	{"asm/ioctls.h",	"TIOCSERGWILD",	0x5454},
 	{"asm/ioctls.h",	"TIOCSERSWILD",	0x5455},
@@ -65,7 +36,6 @@
 	{"asm/ioctls.h",	"TIOCSERSETMULTI",	0x545b},
 	{"asm/ioctls.h",	"TIOCMIWAIT",	0x545c},
 	{"asm/ioctls.h",	"TIOCGICOUNT",	0x545d},
-	{"asm-generic/ioctls.h",	"FIOQSIZE",	0x5460},
 	{"asm/ioctls.h",	"FIOCLEX",	0x6601},
 	{"asm/ioctls.h",	"FIONCLEX",	0x6602},
 	{"asm/ioctls.h",	"FIOASYNC",	0x667d},
diff --git a/linux/s390/ioctlent.h.in b/linux/s390/ioctlent.h.in
index 35cdd81..1e93a0b 100644
--- a/linux/s390/ioctlent.h.in
+++ b/linux/s390/ioctlent.h.in
@@ -19,76 +19,7 @@
 	{"asm/cmb.h",	"BIODASDCMFDISABLE",	0x4421},
 	{"asm/cmb.h",	"BIODASDREADALLCMB",	0x4421},
 	{"asm/dasd.h",	"BIODASDSYMMIO",	0x44f0},
-	{"asm-generic/ioctls.h",	"TCGETS",	0x5401},
-	{"asm-generic/ioctls.h",	"TCSETS",	0x5402},
-	{"asm-generic/ioctls.h",	"TCSETSW",	0x5403},
-	{"asm-generic/ioctls.h",	"TCSETSF",	0x5404},
-	{"asm-generic/ioctls.h",	"TCGETA",	0x5405},
-	{"asm-generic/ioctls.h",	"TCSETA",	0x5406},
-	{"asm-generic/ioctls.h",	"TCSETAW",	0x5407},
-	{"asm-generic/ioctls.h",	"TCSETAF",	0x5408},
-	{"asm-generic/ioctls.h",	"TCSBRK",	0x5409},
-	{"asm-generic/ioctls.h",	"TCXONC",	0x540a},
-	{"asm-generic/ioctls.h",	"TCFLSH",	0x540b},
-	{"asm-generic/ioctls.h",	"TIOCEXCL",	0x540c},
-	{"asm-generic/ioctls.h",	"TIOCNXCL",	0x540d},
-	{"asm-generic/ioctls.h",	"TIOCSCTTY",	0x540e},
-	{"asm-generic/ioctls.h",	"TIOCGPGRP",	0x540f},
-	{"asm-generic/ioctls.h",	"TIOCSPGRP",	0x5410},
-	{"asm-generic/ioctls.h",	"TIOCOUTQ",	0x5411},
-	{"asm-generic/ioctls.h",	"TIOCSTI",	0x5412},
-	{"asm-generic/ioctls.h",	"TIOCGWINSZ",	0x5413},
-	{"asm-generic/ioctls.h",	"TIOCSWINSZ",	0x5414},
-	{"asm-generic/ioctls.h",	"TIOCMGET",	0x5415},
-	{"asm-generic/ioctls.h",	"TIOCMBIS",	0x5416},
-	{"asm-generic/ioctls.h",	"TIOCMBIC",	0x5417},
-	{"asm-generic/ioctls.h",	"TIOCMSET",	0x5418},
-	{"asm-generic/ioctls.h",	"TIOCGSOFTCAR",	0x5419},
-	{"asm-generic/ioctls.h",	"TIOCSSOFTCAR",	0x541a},
-	{"asm-generic/ioctls.h",	"FIONREAD",	0x541b},
-	{"asm-generic/ioctls.h",	"TIOCLINUX",	0x541c},
-	{"asm-generic/ioctls.h",	"TIOCCONS",	0x541d},
-	{"asm-generic/ioctls.h",	"TIOCGSERIAL",	0x541e},
-	{"asm-generic/ioctls.h",	"TIOCSSERIAL",	0x541f},
-	{"asm-generic/ioctls.h",	"TIOCPKT",	0x5420},
-	{"asm-generic/ioctls.h",	"FIONBIO",	0x5421},
-	{"asm-generic/ioctls.h",	"TIOCNOTTY",	0x5422},
-	{"asm-generic/ioctls.h",	"TIOCSETD",	0x5423},
-	{"asm-generic/ioctls.h",	"TIOCGETD",	0x5424},
-	{"asm-generic/ioctls.h",	"TCSBRKP",	0x5425},
-	{"asm-generic/ioctls.h",	"TIOCSBRK",	0x5427},
-	{"asm-generic/ioctls.h",	"TIOCCBRK",	0x5428},
-	{"asm-generic/ioctls.h",	"TIOCGSID",	0x5429},
-	{"asm-generic/ioctls.h",	"TCGETS2",	0x542a},
-	{"asm-generic/ioctls.h",	"TCSETS2",	0x542b},
-	{"asm-generic/ioctls.h",	"TCSETSW2",	0x542c},
-	{"asm-generic/ioctls.h",	"TCSETSF2",	0x542d},
-	{"asm-generic/ioctls.h",	"TIOCGRS485",	0x542e},
-	{"asm-generic/ioctls.h",	"TIOCSRS485",	0x542f},
-	{"asm-generic/ioctls.h",	"TIOCGPTN",	0x5430},
-	{"asm-generic/ioctls.h",	"TIOCSPTLCK",	0x5431},
-	{"asm-generic/ioctls.h",	"TCGETX",	0x5432},
-	{"asm-generic/ioctls.h",	"TIOCGDEV",	0x5432},
-	{"asm-generic/ioctls.h",	"TCSETX",	0x5433},
-	{"asm-generic/ioctls.h",	"TCSETXF",	0x5434},
-	{"asm-generic/ioctls.h",	"TCSETXW",	0x5435},
-	{"asm-generic/ioctls.h",	"TIOCSIG",	0x5436},
-	{"asm-generic/ioctls.h",	"FIONCLEX",	0x5450},
-	{"asm-generic/ioctls.h",	"FIOCLEX",	0x5451},
-	{"asm-generic/ioctls.h",	"FIOASYNC",	0x5452},
-	{"asm-generic/ioctls.h",	"TIOCSERCONFIG",	0x5453},
-	{"asm-generic/ioctls.h",	"TIOCSERGWILD",	0x5454},
-	{"asm-generic/ioctls.h",	"TIOCSERSWILD",	0x5455},
-	{"asm-generic/ioctls.h",	"TIOCGLCKTRMIOS",	0x5456},
-	{"asm-generic/ioctls.h",	"TIOCSLCKTRMIOS",	0x5457},
-	{"asm-generic/ioctls.h",	"TIOCSERGSTRUCT",	0x5458},
-	{"asm-generic/ioctls.h",	"TIOCSERGETLSR",	0x5459},
-	{"asm-generic/ioctls.h",	"TIOCSERGETMULTI",	0x545a},
-	{"asm-generic/ioctls.h",	"TIOCSERSETMULTI",	0x545b},
-	{"asm-generic/ioctls.h",	"TIOCMIWAIT",	0x545c},
-	{"asm-generic/ioctls.h",	"TIOCGICOUNT",	0x545d},
 	{"asm/ioctls.h",	"FIOQSIZE",	0x545e},
-	{"asm-generic/ioctls.h",	"FIOQSIZE",	0x5460},
 	{"asm/chsc.h",	"CHSC_START",	0x6381},
 	{"asm/chsc.h",	"CHSC_INFO_CHANNEL_PATH",	0x6382},
 	{"asm/chsc.h",	"CHSC_INFO_CU",	0x6383},
@@ -116,10 +47,3 @@
 	{"asm/zcrypt.h",	"Z90STAT_PCIXCCMCL3COUNT",	0x7a4c},
 	{"asm/zcrypt.h",	"Z90STAT_CEX2CCOUNT",	0x7a4d},
 	{"asm/zcrypt.h",	"Z90STAT_CEX2ACOUNT",	0x7a4e},
-	{"asm-generic/sockios.h",	"FIOSETOWN",	0x8901},
-	{"asm-generic/sockios.h",	"SIOCSPGRP",	0x8902},
-	{"asm-generic/sockios.h",	"FIOGETOWN",	0x8903},
-	{"asm-generic/sockios.h",	"SIOCGPGRP",	0x8904},
-	{"asm-generic/sockios.h",	"SIOCATMARK",	0x8905},
-	{"asm-generic/sockios.h",	"SIOCGSTAMP",	0x8906},
-	{"asm-generic/sockios.h",	"SIOCGSTAMPNS",	0x8907},
diff --git a/linux/sparc/ioctlent.h.in b/linux/sparc/ioctlent.h.in
index 25b7b4d..008aae0 100644
--- a/linux/sparc/ioctlent.h.in
+++ b/linux/sparc/ioctlent.h.in
@@ -32,76 +32,24 @@
 	{"asm/openpromio.h",	"OPIOCGETNEXT",	0x4f05},
 	{"asm/openpromio.h",	"OPIOCGETCHILD",	0x4f06},
 	{"asm/ioctls.h",	"TCGETA",	0x5401},
-	{"asm-generic/ioctls.h",	"TCGETS",	0x5401},
 	{"asm/ioctls.h",	"TCSETA",	0x5402},
-	{"asm-generic/ioctls.h",	"TCSETS",	0x5402},
 	{"asm/ioctls.h",	"TCSETAW",	0x5403},
-	{"asm-generic/ioctls.h",	"TCSETSW",	0x5403},
 	{"asm/ioctls.h",	"TCSETAF",	0x5404},
-	{"asm-generic/ioctls.h",	"TCSETSF",	0x5404},
-	{"asm-generic/ioctls.h",	"TCGETA",	0x5405},
 	{"asm/ioctls.h",	"TCSBRK",	0x5405},
-	{"asm-generic/ioctls.h",	"TCSETA",	0x5406},
 	{"asm/ioctls.h",	"TCXONC",	0x5406},
 	{"asm/ioctls.h",	"TCFLSH",	0x5407},
-	{"asm-generic/ioctls.h",	"TCSETAW",	0x5407},
 	{"asm/ioctls.h",	"TCGETS",	0x5408},
-	{"asm-generic/ioctls.h",	"TCSETAF",	0x5408},
-	{"asm-generic/ioctls.h",	"TCSBRK",	0x5409},
 	{"asm/ioctls.h",	"TCSETS",	0x5409},
 	{"asm/ioctls.h",	"TCSETSW",	0x540a},
-	{"asm-generic/ioctls.h",	"TCXONC",	0x540a},
-	{"asm-generic/ioctls.h",	"TCFLSH",	0x540b},
 	{"asm/ioctls.h",	"TCSETSF",	0x540b},
 	{"asm/ioctls.h",	"TCGETS2",	0x540c},
-	{"asm-generic/ioctls.h",	"TIOCEXCL",	0x540c},
 	{"asm/ioctls.h",	"TCSETS2",	0x540d},
-	{"asm-generic/ioctls.h",	"TIOCNXCL",	0x540d},
 	{"asm/ioctls.h",	"TCSETSW2",	0x540e},
-	{"asm-generic/ioctls.h",	"TIOCSCTTY",	0x540e},
 	{"asm/ioctls.h",	"TCSETSF2",	0x540f},
-	{"asm-generic/ioctls.h",	"TIOCGPGRP",	0x540f},
-	{"asm-generic/ioctls.h",	"TIOCSPGRP",	0x5410},
-	{"asm-generic/ioctls.h",	"TIOCOUTQ",	0x5411},
-	{"asm-generic/ioctls.h",	"TIOCSTI",	0x5412},
-	{"asm-generic/ioctls.h",	"TIOCGWINSZ",	0x5413},
-	{"asm-generic/ioctls.h",	"TIOCSWINSZ",	0x5414},
-	{"asm-generic/ioctls.h",	"TIOCMGET",	0x5415},
-	{"asm-generic/ioctls.h",	"TIOCMBIS",	0x5416},
-	{"asm-generic/ioctls.h",	"TIOCMBIC",	0x5417},
-	{"asm-generic/ioctls.h",	"TIOCMSET",	0x5418},
-	{"asm-generic/ioctls.h",	"TIOCGSOFTCAR",	0x5419},
-	{"asm-generic/ioctls.h",	"TIOCSSOFTCAR",	0x541a},
-	{"asm-generic/ioctls.h",	"FIONREAD",	0x541b},
 	{"asm/ioctls.h",	"TIOCLINUX",	0x541c},
-	{"asm-generic/ioctls.h",	"TIOCCONS",	0x541d},
 	{"asm/ioctls.h",	"TIOCGSERIAL",	0x541e},
 	{"asm/ioctls.h",	"TIOCSSERIAL",	0x541f},
-	{"asm-generic/ioctls.h",	"TIOCPKT",	0x5420},
-	{"asm-generic/ioctls.h",	"FIONBIO",	0x5421},
-	{"asm-generic/ioctls.h",	"TIOCNOTTY",	0x5422},
-	{"asm-generic/ioctls.h",	"TIOCSETD",	0x5423},
-	{"asm-generic/ioctls.h",	"TIOCGETD",	0x5424},
 	{"asm/ioctls.h",	"TCSBRKP",	0x5425},
-	{"asm-generic/ioctls.h",	"TIOCSBRK",	0x5427},
-	{"asm-generic/ioctls.h",	"TIOCCBRK",	0x5428},
-	{"asm-generic/ioctls.h",	"TIOCGSID",	0x5429},
-	{"asm-generic/ioctls.h",	"TCGETS2",	0x542a},
-	{"asm-generic/ioctls.h",	"TCSETS2",	0x542b},
-	{"asm-generic/ioctls.h",	"TCSETSW2",	0x542c},
-	{"asm-generic/ioctls.h",	"TCSETSF2",	0x542d},
-	{"asm-generic/ioctls.h",	"TIOCGRS485",	0x542e},
-	{"asm-generic/ioctls.h",	"TIOCSRS485",	0x542f},
-	{"asm-generic/ioctls.h",	"TIOCGPTN",	0x5430},
-	{"asm-generic/ioctls.h",	"TIOCSPTLCK",	0x5431},
-	{"asm-generic/ioctls.h",	"TCGETX",	0x5432},
-	{"asm-generic/ioctls.h",	"TCSETX",	0x5433},
-	{"asm-generic/ioctls.h",	"TCSETXF",	0x5434},
-	{"asm-generic/ioctls.h",	"TCSETXW",	0x5435},
-	{"asm-generic/ioctls.h",	"TIOCSIG",	0x5436},
-	{"asm-generic/ioctls.h",	"FIONCLEX",	0x5450},
-	{"asm-generic/ioctls.h",	"FIOCLEX",	0x5451},
-	{"asm-generic/ioctls.h",	"FIOASYNC",	0x5452},
 	{"asm/ioctls.h",	"TIOCSERCONFIG",	0x5453},
 	{"asm/ioctls.h",	"TIOCSERGWILD",	0x5454},
 	{"asm/ioctls.h",	"TIOCSERSWILD",	0x5455},
@@ -113,7 +61,6 @@
 	{"asm/ioctls.h",	"TIOCSERSETMULTI",	0x545b},
 	{"asm/ioctls.h",	"TIOCMIWAIT",	0x545c},
 	{"asm/ioctls.h",	"TIOCGICOUNT",	0x545d},
-	{"asm-generic/ioctls.h",	"FIOQSIZE",	0x5460},
 	{"asm/watchdog.h",	"WIOCSTART",	0x570a},
 	{"asm/watchdog.h",	"WIOCSTOP",	0x570b},
 	{"asm/watchdog.h",	"WIOCGSTAT",	0x570c},
-- 
1.8.0

